scripts/casefold_db.py

```diff
@@ -38,18 +80,67 @@ def calculate_lookup_hash(sydent, address):
     return lookup_hash
 
 
+def sendEmailWithBackoff(
+    sydent: Sydent,
+    address: str,
+    mxid: str,
+    backoff: int,
+    test: bool = False,
+    attempts: int = 0,
+) -> None:
+    """Send an email with exponential backoff - that way we don't stop sending halfway
+    through if the SMTP server rejects our email (e.g. because of rate limiting).
+
+    Setting test to True disables the logging.
+    """
+    if attempts == MAX_ATTEMPTS_FOR_EMAIL:
+        raise CantSendEmailException()
```

Comment:
I think this code would be simpler if we used a loop rather than recursion, that way we can get rid of both the `backoff` and `attempts` args and make them local.

It should just be a case of sticking a `for _ in range(MAX_ATTEMPTS_FOR_EMAIL):` in front.